streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -118,7 +127,7 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
 
         topology.addProcessor(name, new KStreamMap<>(mapper), this.name);
 
-        return new KStreamImpl<>(topology, name, null);
+        return new KStreamImpl<>(topology, name, null, null, null);
```

Comment:
I remember that Flink can infer return types for some cases even for map function, for example:

```
public class AppendOne<T> extends MapFunction<T, Tuple2<T, Long>> { public Tuple2<T, Long> map(T value) { return new Tuple2<T, Long>(value, 1L); } }
```

Do you know how it can be achieved?